js/util/proxy.js

```diff
@@ -0,0 +1,75 @@
+if (typeof require !== 'undefined') {
+  var settings = require('util/settings/settings.js')
+}
+const { BrowserWindow } = require('electron')
+const { session } = require('electron')
+
+var proxy = {
+  rules: {
+    proxyBypassRules: "localhost"
+  },
+  initialize: function () {
+    settings.listen("proxyMode", function(value){
+      if (value == 'no-proxy') {
+        proxy.changeHandler("")
+      }
+      else {
+        var proxyRules = {}
+        settings.listen('proxies', function(proxy){
+          if ( proxy.httpHost ) {
+            var proxyString
+            if ( !(proxy.httpPort == 0) )
+              proxyString = "http://" + proxy.httpHost + ":" + proxy.httpPort
+            else
+              proxyString = "http://" + proxy.httpHost
+            proxyRules.httpProxy = proxyString
+          }
+          if ( proxy.sslHost ) {
+            var proxyString
+            if ( !(proxy.sslPort == 0) )
+              proxyString = "https://" + proxy.sslHost + ":" + proxy.sslPort
+            else
+              proxyString = "https://" + proxy.sslHost
+            proxyRules.sslProxy = proxyString
+          }
+          if ( proxy.ftpHost ) {
+            var proxyString
+            if ( !(proxy.ftpPort == 0) )
+              proxyString = "http://" + proxy.ftpHost + ":" + proxy.ftpPort
+            else
+              proxyString = "http://" + proxy.ftpHost
+            proxyRules.ftpProxy = proxyString
+          }
+          if ( proxy.socksHost ) {
+            var proxyString
+            if( proxy.socksMode == "socks4" ) {
+              proxyString = "socks4://"
+            }
+            else {
+              proxyString = "socks5://"
+            }
+
+            if ( !(proxy.socksPort == 0) )
+              proxyString += proxy.socksHost + ":" + proxy.socksPort
+            else
+              proxyString += proxy.socksHost
+            proxyRules.socksProxy = proxyString
+          }
+        })
+        proxy.changeHandler( Object.values(proxyRules).join(";") )
+      }
+    })
+  },
+  changeHandler: function(proxyRules) {
+    proxy.rules.proxyRules = proxyRules
+    window.dispatchEvent(new CustomEvent("proxy-change", {
+      detail: proxy.rules
+    }))
+  }
+}
+
+if (require.main === module) {

```

Comment:
`proxy.js` isn't ever loaded not as a module, so this check shouldn't be necessary.

Reply:
Yes I should remove this, as I first wanted to add it in legacy mode but latter changed it.